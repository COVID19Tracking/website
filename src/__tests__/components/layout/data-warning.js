import React from 'react'
import renderer from 'react-test-renderer'
import * as Gatsby from 'gatsby'
import { StaticQuery } from 'gatsby'
import DataWarning from '~components/layout/header/data-warning'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')

describe('Components : Layout : Data warning', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })
  it('is hidden if the snippet is empty', () => {
    useStaticQuery.mockImplementation(() => ({
      contentfulSnippet: {
        contentful_id: '1234',
        name: 'data-warning',
        slug: 'data-warning',
        content: {
          content: '',
        },
      },
    }))
    const tree = renderer.create(<DataWarning />).toJSON()
    expect(tree).toBeNull()
  })

  it('is displayed if there is content', () => {
    useStaticQuery.mockImplementation(() => ({
      contentfulSnippet: {
        contentful_id: '1234',
        name: 'data-warning',
        slug: 'data-warning',
        content: {
          content: 'There are some data issues and **we are on it**.',
        },
      },
    }))
    const tree = renderer.create(<DataWarning />).toJSON()
    expect(tree).not.toBeNull()
    expect(tree).toMatchSnapshot()
  })
})
