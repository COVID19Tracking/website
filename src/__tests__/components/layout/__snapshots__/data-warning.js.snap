// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Layout : Data warning is displayed if there is content 1`] = `
<div
  className="warning"
>
  <div
    className="container container"
  >
    <div
      className="full"
    >
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "There are some data issues and <strong>we are on it</strong>.",
          }
        }
        role="alert"
      />
    </div>
  </div>
</div>
`;
